[Chai Wan Lin] iP

chaiwanlin opened the pull request:
# Pepper Jack
> Father, hold me close, don't let me drown
> I know you won't
> - Kanye West, Hurricane

_**Pepper Jack**_ helps you track your tasks! It's

* text-based
* **easy** to use and learn

How to use:

1. Download from [here](https://github.com/chaiwanlin/ip/releases/tag/v0.1)
2. Double-click on `.jar` file
3. Try it out!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon:wink:) 

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```javascript
public static void main(String[] args) {
        new Duke("data/tasks.txt").run();
}
```

Commit:
Add Gradle support

chaiwanlin committed:
Level-1

chaiwanlin committed:
Level-2

chaiwanlin committed:
Level-3

chaiwanlin committed:
Level-4

chaiwanlin committed:
A-TextUiTesting

chaiwanlin committed:
Level-5

chaiwanlin committed:
Level-6

chaiwanlin committed:
Add save function

chaiwanlin committed:
Add date recognition

chaiwanlin committed:
Merge branch 'branch-Level-8'

chaiwanlin committed:
Add new subclasses for OOP

chaiwanlin committed:
Change to duke package

chaiwanlin committed:
Update storage of tasks to new class

chaiwanlin committed:
Update storage class functionality

chaiwanlin committed:
Update storage class

chaiwanlin committed:
Update Ui class functionality

chaiwanlin committed:
Add Ui class functionality

chaiwanlin committed:
Update packaging

aaronloh852 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,369 @@
+                        throw new DukeException("Deadline when brah??\n\t" + Command.DEADLINE.help_text);
+                    }
+                    // Make sure date format is correct
+                    try {
+                        LocalDate.parse(desc_date[1]);
+                    }
+                    catch (DateTimeParseException e) {
+                        throw new DukeException("Follow the gahdam format!!\n\t" + Command.DEADLINE.help_text);
+                    }
+                    return Command.DEADLINE;
+                }
+                case "event": {
+                    // Make sure there is key phrase after to record as task
+                    if (key_phrases == 1) {
+                        throw new DukeException("What event brah??\n\t" + Command.EVENT.help_text);
+                    }
+                    // Makes sure key phrase after contains /at descriptor to record datetime
+                    String[] desc_date = input_split[1].split(" /at ", 2);
+                    if (desc_date.length != 2) {
+                        throw new DukeException("Event when brah??\n\t" + Command.EVENT.help_text);
+                    }
+                    // Make sure date format is correct
+                    try {
+                        LocalDate.parse(desc_date[1]);
+                    }
+                    catch (DateTimeParseException e) {
+                        throw new DukeException("Follow the gahdam format!!\n\t" + Command.EVENT.help_text);
+                    }
+                    return Command.EVENT;
+                }
+                default: {
+                    // If invalid command, show all help text to user
+                    String all_help_text = "";
+                    for (Command c : Command.values()) {
+                        all_help_text = all_help_text + "\t" + c.help_text + "\n";
+                    }
+                    throw new DukeException("What you sayin brah! You better start makin sense or else...\n" +
+                            "*Speak in words Pepper Jack can understand*\n" + all_help_text);
+                }
+            }
```
I think switch statements should not have indentation for the cases?

aaronloh852 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,369 @@
+                            String desc = user_input.split(" ", 2)[1];
+                            int index = Integer.parseInt(desc) - 1;
+                            Task t = lst.get(index);
+                            t.setDone();
+                            System.out.print(reply("Noice! Pepper Jack marked this task as done:\n\t" + t));
+                            break;
+                        }
+                        case DELETE: {
+                            // Delete task
+                            String desc = user_input.split(" ", 2)[1];
+                            int index = Integer.parseInt(desc) - 1;
+                            Task t = lst.remove(index);
+                            System.out.print(showTasksReply(c, "Aights! Pepper Jack deleted this task:\n\t" + t));
+                            break;
+                        }
+                        case TODO: {
+                            // Add new to do
+                            String desc = user_input.split(" ", 2)[1];
+                            Task t = new Todo(desc);
+                            lst.add(t);
+                            System.out.print(showTasksReply(c, t.toString()));
+                            break;
+                        }
+                        case DEADLINE: {
+                            // Add new deadline
+                            String desc_date = user_input.split(" ", 2)[1];
+                            Task t = Deadline.build(desc_date);
+                            lst.add(t);
+                            System.out.print(showTasksReply(c, t.toString()));
+                            break;
+                        }
+                        case EVENT: {
+                            // Add new event
+                            String desc_date = user_input.split(" ", 2)[1];
+                            Task t = Event.build(desc_date);
+                            lst.add(t);
+                            System.out.print(showTasksReply(c, t.toString()));
+                            break;
+                        }
+                    }
```
Same comment about switches as above

aaronloh852 reviewed (commented):
I think the code is clean and well organised! Maybe you can add header comments to the javadoc comments?

chaiwanlin committed:
Update .gitignore

chaiwanlin committed:
Add new test cases

chaiwanlin committed:
Update text-ui-test

chaiwanlin committed:
Update ByeCommand

chaiwanlin committed:
Update Command

chaiwanlin committed:
Merge branch 'branch-Level-8'

chaiwanlin committed:
Add tag

chaiwanlin committed:
Add javadoc comments

chaiwanlin committed:
Comply with coding standard

chaiwanlin committed:
Add find command implementation

chaiwanlin committed:
Merge branch 'branch-A-CodingStandard'

chaiwanlin committed:
Merge branch 'branch-Level-9'

tau-bar commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,175 @@
+     * @return DoneCommand to execute.
+     * @throws DukeException If there are no arguments given or argument is not an integer.
+     */
+    public static Command prepareDone(String args) throws DukeException {
```
I think the naming of this method as well as the parameter could be made clearer, something like:
```suggestion
    public static Command returnDoneCommand(String indexOfTask) throws DukeException {
```

tau-bar commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,175 @@
+package duke;
```
I think the naming of the methods and method parameters could be made clearer, I did an example on line 66.

tau-bar commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,110 @@
+     * @throws DukeException If file is not found.
+     * Should not happen since existence of file is checked.
+     */
+    public TaskList load() throws DukeException {
```
Again, I think the naming of this method could be made more explicit to demonstrate it's function, like: 
```suggestion
    public TaskList loadTasksFromStorage() throws DukeException {
```

tau-bar commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,43 @@
+     *                  '[description] (by: [MMM d yyyy]' format.
+     * @return Deadline from string input with description and date.
+     */
+    public static Deadline build(String desc_date) {
```
I think you should change this to be camel case.
```suggestion
    public static Deadline build(String descriptionAndDate) {
```

tau-bar commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,47 @@
+     *                  '[description] (by: [MMM d yyyy]' format.
+     * @return Event from string input with description and date.
+     */
+    public static Event build(String desc_date) {
```
I think this should be camel case, as mentioned above.

tau-bar commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    /** Sets isDone to given boolean */
+    public void setIsDone(boolean b) {
+        isDone = b;
+    }
```
Could this be better if it just marks the task as done instead of having to pass in a boolean parameter?
```suggestion
    public void setDone() {
        isDone = true;
    }
```

tau-bar reviewed (commented):
LGTM, just some minor issues with the naming of methods and parameters, as mentioned [here](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#use-name-to-explain). I am impressed by the depth of the documnentation and the cleanliness of the code. Other than that, I like how you added your own flavour to the bot. 😄

chaiwanlin committed:
Merge branch 'add-gradle-support' of https://github.com/chaiwanlin/ip

chaiwanlin committed:
Configure gradle to run Duke and unit tests

chaiwanlin committed:
Add basic GUI

chaiwanlin committed:
Add DialogBox class

chaiwanlin committed:
Update to FXML

chaiwanlin committed:
Improve UI

chaiwanlin committed:
Update build.gradle to build JAR

chaiwanlin committed:
Update header comments

chaiwanlin committed:
Add assertions to Parser and Duke

Arguments passed to prepare Commands and response generated
may be null.

Adding assertions ensures that the arguments passed to prepare
Commands and response given to users cannot be null.

chaiwanlin committed:
Add assertion to Duke

Previous commit included assertion to check for empty response
but not null.

Adding null assertion for response ensure there is a response
returned.

chaiwanlin committed:
Improve code quality standard

chaiwanlin committed:
Merge pull request #1 from chaiwanlin/branch-A-Assertions

Add assertions to Duke and Parser classes.

chaiwanlin committed:
Update header comments

chaiwanlin committed:
Merge pull request #2 from chaiwanlin/branch-A-CodeQuality

Improve code quality according to standard

chaiwanlin committed:
Prevent user from adding duplicate tasks

chaiwanlin committed:
Update exit command for GUI

chaiwanlin committed:
Improve GUI

chaiwanlin committed:
Merge branch 'branch-C-DetectDuplicates'

chaiwanlin committed:
Add title to GUI window

chaiwanlin committed:
Merge branch 'branch-A-BetterGui'

chaiwanlin committed:
Update to include welcome message

chaiwanlin committed:
Debug loading of saved tasks

chaiwanlin committed:
Merge branch 'branch-A-BetterGui'

chaiwanlin committed:
Update to save task on close window

chaiwanlin committed:
Merge branch 'branch-A-BetterGui'

chaiwanlin committed:
Remove print statement in TaskList

chaiwanlin committed:
Merge branch 'master' into branch-A-BetterGui

chaiwanlin committed:
Improve GUI text field

chaiwanlin committed:
Change GUI font to Helvetica

chaiwanlin committed:
Add speech bubble to GUI

chaiwanlin committed:
Update displayPicure format and margins

chaiwanlin committed:
Add icon for GUI

chaiwanlin committed:
Add representative screenshot in docs

chaiwanlin committed:
Add User Guide and change Ui.png
